core-bundle/src/SimpleTokens/Parser.php

```diff
@@ -0,0 +1,231 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\SimpleTokens;
+
+use Psr\Log\LoggerInterface;
+use Psr\Log\LogLevel;
+use Symfony\Component\ExpressionLanguage\ExpressionLanguage;
+
+class Parser
+{
+    /**
+     * @var LoggerInterface|null
+     */
+    private $logger;
+
+    /**
+     * @var ExpressionLanguage
+     */
+    private $expressionLanguage;
+
+    public function __construct(LoggerInterface $logger = null, ExpressionLanguage $expressionLanguage = null)
+    {
+        if (null === $expressionLanguage) {
+            $expressionLanguage = new ExpressionLanguage();
+        }
+
+        // Disable `constant()` function for security reasons.
+        $expressionLanguage->register(
+            'constant',
+            static function () {
+                return "throw new \\InvalidArgumentException('Cannot use the constant() function in the expression for security reasons.');";
+            },
+            static function (): void {
+                throw new \InvalidArgumentException('Cannot use the constant() function in the expression for security reasons.');
+            }
+        );
+
+        $this->expressionLanguage = $expressionLanguage;
+        $this->logger = $logger;
```

Comment:
Guess we could just make that service logger aware instead of having a constructor attribute?

Reply:
eh, sure!